app/controllers/api/v1/search_controller.rb

```diff
@@ -1,31 +1,22 @@
 class Api::V1::SearchController < ApplicationController
   def find_merchant
-    if params[:name] && params[:name].present?
-      merchant =
-      Merchant.where('name Ilike ?', "%#{params[:name]}%")
-      .order('LOWER(name)')
-      .first
-
-      if merchant
-        render json: MerchantSerializer.new(merchant)
-      else
-        blank_merchant = Merchant.create
-        render json: MerchantSerializer.new(blank_merchant), status: 400
-      end
+    if params[:name]&.present?
+      merchant = Merchant.find_match_by_name(params[:name]).first
+      render json: MerchantSerializer.new(merchant) if merchant.present?
+      render json: MerchantSerializer.new(Merchant.create), status: :bad_request if merchant.nil?
+      # I would have just created a no found error for this
+      # render json: { error: 'Merchant not found' }.to_json, status: :not_found if merchant.nil?
     else
-      render json: { error: 'Please include name param' }.to_json, status: 400
+      render json: { error: 'Please include name param' }.to_json, status: :bad_request
     end
   end
 
   def find_items
-    if params[:name] && params[:name].present?
-      items = Item.where('name Ilike ?', "%#{params[:name]}%")
-                  .order('LOWER(name)')
-      if items
-        render json: ItemSerializer.new(items)
-      end
+    if params[:name]&.present?
+      items = Item.find_match_by_name(params[:name])
```

Comment:
find_match_by_name?!? How fricken cool is that? What a great refactor!

Would love to talk about nested renders/error handling a little bit with you

Reply:
Nested ifs are a pain to most people. You are doing great!